IPv6 and Fatal exception occurred in thread #0: A call to an external process failed

While using IPv6 on osx I get the following error which causes the core to close and then the program to quit. I have tested this on the 2.2.

```
2019-06-02T09:51:43.615664 [ERROR] Fatal exception occurred in thread #0: A call to an external process failed
```

I did some debugging myself and added the following to `./libs/executeplus/src/posix_system.cpp`

```
void checked_execute(const std::vector<std::string>& args, const std::map<std::string, std::string>& env, std::ostream* output)
  {
    std::string s = "";
    for (unsigned i=0; i < args.size(); i++) {
      s += args[i];
      s += ", ";
    }
    s += "\n";
    std::cout << s;

    if (execute(args, env, output) != 0)
    {
      throw boost::system::system_error(make_error_code(executeplus_error::external_process_failed));
    }
  }
```

This was able to tell me what the external program that failed was.

Which was
```
/sbin/route -n get XXXX:XXXX:XXXX:XXXX:XXXX:XXXX:XXXX:XXXX
```
Trying the command myself

```
/sbin/route -n get XXXX:XXXX:XXXX:XXXX:XXXX:XXXX:XXXX:XXXX
route: bad address: XXXX:XXXX:XXXX:XXXX:XXXX:XXXX:XXXX:XXXX
```

So this seems to be not how you use /sbin/route for ipv6.

Checking the man pages!

To successfully run this command it should be
```
/sbin/route -n get -inet6 XXXX:XXXX:XXXX:XXXX:XXXX:XXXX:XXXX:XXXX
```

So two issues:
1. Fix /sbin/route for ipv6
2. Could you add more debug info than just `A call to an external process failed` I spent way to long finding the line of code needed to just tell me that it failed to run `/sbin/route`. Perhaps just adding it to the debug build would be enough.

--- a/libs/asiotap/src/posix/posix_route_manager.cpp
+++ b/libs/asiotap/src/posix/posix_route_manager.cpp
@@ -64,7 +64,8 @@ namespace asiotap
 	posix_route_manager::route_type posix_route_manager::get_route_for(const boost::asio::ip::address& host)
 	{
 #ifdef MACINTOSH
-		const std::vector<std::string> real_args { "/sbin/route", "-n", "get", boost::lexical_cast<std::string>(host) };
+		const std::string inet = host.is_v6() ? "-inet6" : "-inet";
+		const std::vector<std::string> real_args { "/sbin/route", "-n", "get", inet, boost::lexical_cast<std::string>(host) };
 
 		std::stringstream ss;
 		executeplus::checked_execute(real_args, executeplus::get_current_environment(), &ss);
@@ -197,7 +198,8 @@ namespace asiotap
 		const std::string net_host = is_unicast(dest) ? "-host" : "-net";
 #ifdef MACINTOSH
 		const std::string command = action == route_action::add ? "add" : "delete";
-		const std::vector<std::string> real_args { "/sbin/route", "-n", command, net_host, boost::lexical_cast<std::string>(dest), "-interface", interface };
+		const std::string inet = boost::apply_visitor(ip_network_address_ip_address_visitor(), dest).is_v6() ? "-inet6" : "-inet";
+		const std::vector<std::string> real_args { "/sbin/route", "-n", command, net_host, inet, boost::lexical_cast<std::string>(dest), "-interface", interface };
 #else
 		const std::string command = action == route_action::add ? "add" : "del";
 		const std::vector<std::string> real_args { "/sbin/route", "-n", command, net_host, boost::lexical_cast<std::string>(dest), "dev", interface };

